gate-plugins/src/main/kotlin/com/netflix/spinnaker/gate/plugins/web/publish/PluginPublishController.kt

```diff
@@ -27,32 +29,42 @@ import okhttp3.MultipartBody
 import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.RequestBody
+import org.springframework.http.MediaType.MULTIPART_FORM_DATA_VALUE
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestMapping
-import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RequestPart
 import org.springframework.web.bind.annotation.RestController
 import org.springframework.web.multipart.MultipartFile
 
 @RestController
-@RequestMapping("/plugins/upload")
-class PluginBinaryController(
+@RequestMapping("/plugins/publish")
+class PluginPublishController(
+  private val pluginService: PluginService,
   private val okHttpClient: OkHttpClient,
-  private val serviceConfiguration: ServiceConfiguration
+  serviceConfiguration: ServiceConfiguration
 ) {
 
+  private val front50Url = serviceConfiguration.getServiceEndpoint("front50").url
+
   @SneakyThrows
-  @ApiOperation(value = "Upload a plugin binary")
-  @PostMapping("/{pluginId}/{pluginVersion}")
-  fun publishBinary(
-    @RequestParam("plugin") body: MultipartFile,
+  @ApiOperation(value = "Publish a plugin binary and the plugin info metadata.")
+  @PostMapping("/{pluginId}/{pluginVersion}", consumes = [MULTIPART_FORM_DATA_VALUE])
+  fun publishPlugin(
+    @RequestPart("plugin") body: MultipartFile,
+    @RequestPart("pluginInfo") pluginInfo: SpinnakerPluginInfo,
     @PathVariable pluginId: String,
-    @PathVariable pluginVersion: String,
-    @RequestParam sha512sum: String
+    @PathVariable pluginVersion: String
   ) {
+    pluginService.verifyPluginInfo(pluginInfo, pluginId)
     val bytes = body.bytes
-    verifyChecksum(bytes, sha512sum)
-    uploadToFront50(pluginId, pluginVersion, sha512sum, bytes)
+    val release = pluginService.getReleaseByVersion(pluginInfo, pluginVersion)
+
+    verifyChecksum(bytes, release.sha512sum)
+    uploadToFront50(pluginId, pluginVersion, release.sha512sum, bytes)
+
+    release.url = "$front50Url/pluginBinaries/$pluginId/$pluginVersion"
+    pluginService.upsertPluginInfo(pluginInfo)
```

Comment:
Added some verifying of the provided plugin info, as well as adding the correct front50 binary url to the release.